qiskit/transpiler/passes/synthesis/high_level_synthesis.py

```diff
@@ -110,9 +128,10 @@ def run(self, dag: DAGCircuit) -> DAGCircuit:
         Raises:
             TranspilerError: when the specified synthesis method is not available.
         """
-
         hls_plugin_manager = HighLevelSynthesisPluginManager()
 
+        dag_bit_indices = {bit: i for i, bit in enumerate(dag.qubits)}
```

Comment:
Lol, I think every single pass has this line of code, we really should just bake this into the dagcircuit

Reply:
I opened an issue to track this here: https://github.com/Qiskit/qiskit-terra/issues/9389